aten/src/ATen/test/cuda_distributions_test.cu

```diff
@@ -0,0 +1,77 @@
+#include <gtest/gtest.h>
+
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+#include <curand.h>
+#include <curand_kernel.h>
+#include <curand_philox4x32_x.h>
+
+__device__ void random_float(float* x) {
+  for(int i=0; i < 4; i++) {
+    curandStatePhilox4_32_10_t state;
+    curand_init(
+            123,
+            i,
+            4,
+            &state);
+    auto ret = curand_uniform4(&state);
+    x[i] = ret.x;
+  }
+}
+
+
+__global__ void myKernel(float* x) {
+  random_float(x);
+}
+
+TEST(DistributionsTest, TestPhiloxIncrement) {
+  // Test Description:
+  //   In Distributions.cu we mentioned that philox increment
+  //   should be at least the number of curand() random numbers used in
+  //   each thread. In this test, we make sure that uniform_ correctly
+  //   increments philox and doesn't reuse randoms from previous calls.
+  //    - We check that by first getting 4 randoms from uniform_.
```

Comment:
Should we be testing for other numbers of generated randoms, or is that not useful?

Reply:
Yes, that could be helpful. I'll add some more cases.